packages/common/src/store/collectibles/slice.ts

```diff
@@ -0,0 +1,70 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { Chain, Collectible, ID } from '../../models'
+
+import { Metadata } from '@metaplex-foundation/mpl-token-metadata'
+import { Nullable } from '../../utils'
+
+export type CollectiblesState = {
+  userCollectibles: {
+    [id: ID]: {
+      [Chain.Eth]: Collectible[],
+      [Chain.Sol]: Collectible[]
+    }
+  },
+  solCollections: {
+    [mint: string]: (Metadata & { imageUrl: Nullable<string> })
+  }
+}
+
+const initialState: CollectiblesState = {
+  userCollectibles: {},
+  solCollections: {}
+}
+
+type UpdateUserCollectiblesPayload = {
+  userId: ID,
+  userCollectibles: Collectible[]
+}
+
+type UpdateSolCollectionsPayload = {
+  metadatas: { [mint: string]: (Metadata & { imageUrl: Nullable<string> }) }
+}
+
+const slice = createSlice({
+  name: 'collectibles',
+  initialState,
+  reducers: {
+    updateUserEthCollectibles: (state, action: PayloadAction<UpdateUserCollectiblesPayload>) => {
+      state.userCollectibles[action.payload.userId] = {
+        ...state.userCollectibles[action.payload.userId],
+        [Chain.Eth]: action.payload.userCollectibles
+      }
```

Comment:
nit: could just do bc immer

```state.userCollectibles[userId][Chain.Eth] = action.payload.userCollectibles```

Reply:
this `state.userCollectibles[userId]` may be undefined